src/attendees/attendee.py

```diff
@@ -60,26 +100,114 @@ def has_attribute(self, key: Hashable) -> bool:
         Returns:
             bool: True if Attendee has attribute and vice-versa.
         """
-        raise NotImplementedError
-
+    @property
+    def has_fname(self):    
+        return self._fname == None
+    
+    @property
+    def has_lname(self):    
+        return self._lname == None # already gives a boolean, if statements always accept boolean
+
+    # if condtion, if not condition (i.e. else)
+    
+    @property
+    def has_email(self):    
+        return self._email == None
+
+    @property
+    def has_file_url(self):    
+        return self._file_url == None
+
+    @property
+    def has_file_path(self):    
+        return self._file_url == None
+    
     def set_attribute(self, key: Hashable, value: Any,
-                     verifyFunc: VerifyFunc = None):
+                     verifyFunc: VerifyFunc = None): #verifyFunc is a function, a callback - the what, for miscellaneous, 
+                    # i wont write callback, will use it
         """
         Set an arbitrary attribute of the attendee.
 
-        Will add the attribute if it doesn't exist.
+        Will add the attribute if it doesn't exist. First Checks if `verifyFunc`
+        is `None`. If `None`, do not verify value, any value is validOnce verified,
+        set misc attribute using `key`.
 
         Args:
             key (Hashable): The name of the attribute.
             value (Any): The attribute value.
             verifyFunc (Callable[[Hashable, Any], bool]): A function to verify
-            the value being set, if let as None, will not verify value.
+            the value being set, if let as `None`, will not verify value. If
+            `True`, value is valid and vice-versa.
         
         Raises:
-            TypeError if key is not hashable.
-            ValueError if key is "fname", "lname", or "email"
+            TypeError: if `key` is not hashable.
+            ValueError: if `key` is "fname", "lname", or "email"
+            ValueError: if verifyFunc returns `False`
         """
-        raise NotImplementedError
+        if verifyFunc is not None: # if None, dont need to verify
+            verifyFunc(key, value) # need to make sure its correct
+            if not verifyFunc(key, value):
+                # raise error - value is not valid fo this key
+                raise NotImplementedError
+
+        self._misc[key] = value
+
+    @classmethod
+    def checkString(cls, value):    # checks if string and if empty
+        # cls, allows u to use func without creating object. MyClass.func()
```

Comment:
This should be a static method, not a class method as nothing else inside this class is used in this method.